test/items.html

```diff
@@ -414,6 +422,97 @@
           expect(menuComponents()[0].getAttribute('aria-expanded')).to.equal('false');
         });
       });
+
+      describe('theme propagation', () => {
+        let subMenu2;
+
+        beforeEach(done => {
+          rootMenu = fixture('theme');
+          rootMenu.openOn = OPENEVENT;
+          target = rootMenu.firstElementChild;
+          rootMenu.items = [
+            {text: 'foo-0', children:
+              [
+                {text: 'foo-0-0'},
+                {text: 'foo-0-1', children: [
+                  {text: 'foo-0-1-0'}
+                ]},
+              ]
+            },
+            {text: 'foo-1'}
+          ];
+          open();
+          flush(() => {
+            open(menuComponents()[0]);
+            subMenu = getSubMenu();
+            flush(() => {
+              open(menuComponents(subMenu)[1]);
+              subMenu2 = getSubMenu();
+              flush(done);
+            });
+          });
+        });
+
+        function themeAttributeEquals(theme, el, overlay = false, listBox = false, items = false) {
+          const elOverlay = el.$.overlay;
+          const elListBox = elOverlay.querySelector('vaadin-context-menu-list-box');
+          const elItems = Array.from(elListBox.querySelectorAll('vaadin-context-menu-item'));
+
+          if (overlay) {
+            return elOverlay.getAttribute('theme') === theme;
+          } else if (listBox) {
+            return elListBox.getAttribute('theme') === theme;
+          } else if (items) {
+            return elItems.filter(item => item.getAttribute('theme') !== theme).length === 0;
+          }
+
+          return el.getAttribute('theme') === theme;
+        }
+
+        ['propagate', 'remove'].forEach(condition => {
```

Comment:
This iterator only makes the test suite messy. Would be much easier to follow if these were separate tests. I think you only need 2-3 anyway (see the above comment).

Reply:
Updated. Now there are only 2 tests.